app/controllers/lead_providers/your_schools_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class YourSchoolsController < ::LeadProviders::BaseController
+    before_action :set_lead_provider
+
+    def index
+      @cohorts ||= @lead_provider.cohorts
+      @selected_cohort = if params[:selected_cohort_id]
+                           Cohort.find(params[:selected_cohort_id])
+                         else
+                           Cohort.find_by(start_year: Time.zone.today.year)
```

Comment:
Should this be `@cohorts.find` and `@cohorts.find_by`?

Reply:
Yes - if there is no selected cohort id passed it will default it for this year cohort

Reply:
I mean it should only be possible to select from the cohorts of a lead provider, not all cohorts. `Cohort` becomes `@cohorts`

Reply:
It would be good to use a Pundit policy scope for this perhaps?

Reply:
Good point - the lead provider should see only their own Cohorts

Reply:
Done